sdk/cosmosdb/cosmos/src/globalEndpointManager.ts

```diff
@@ -200,24 +200,27 @@ export class GlobalEndpointManager {
       // if we are not able to get that info from any endpoints
     } catch (err) {
       // TODO: Tracing
-    }
 
-    if (this.preferredLocations) {
-      for (const location of this.preferredLocations) {
-        try {
-          const locationalEndpoint = GlobalEndpointManager.getLocationalEndpoint(
-            this.defaultEndpoint,
-            location
-          );
-          const options = { urlConnection: locationalEndpoint };
-          const { resource: databaseAccount } = await this.readDatabaseAccount(options);
-          if (databaseAccount) {
-            return databaseAccount;
+      if (this.preferredLocations) {
```

Comment:
this is a change in behavior, this code needs to run even if we don't catch an error